### PR TITLE
optimization: optimize a clear interaction for content-type

### PR DESCRIPTION
**Pull request by workman-Lu**
Content-type is not be promise from RFC2616 7.2.1.
it may be need a clear describe info:)

### CODE REVIEW

**Comment by workman-Lu** [comment]
optimize Content-type reason of error

**Review by Windfarer — CHANGES_REQUESTED** [review]

File: transport/http/codec.go [hunk 1]
```diff
@@ -77,5 +77,5 @@ func CodecForRequest(r *http.Request, name string) (encoding.Codec, bool) {
 			return codec, true
 		}
 	}
-	return encoding.GetCodec("json"), false
+	return encoding.GetCodec("UnregisterType"), false
```
[the remark this answers is not in the document]
Comment (workman-Lu) [reply transport/http/codec.go]: sorry about that be overlooked others call ,all right now :).

File: transport/http/codec.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ type EncodeErrorFunc func(http.ResponseWriter, *http.Request, error)
 func DefaultRequestDecoder(r *http.Request, v interface{}) error {
 	codec, ok := CodecForRequest(r, "Content-Type")
 	if !ok {
-		return errors.BadRequest("CODEC", r.Header.Get("Content-Type"))
+		return errors.BadRequest("CODEC unregister Content-Type:", r.Header.Get("Content-Type"))
```
Comment (tonybase) [inline transport/http/codec.go:28]: should be set in message

Comment (workman-Lu) [reply transport/http/codec.go:28]: > should be set in message

There seems to be no other way for me to judge the existence of this problem from the outside of the function:)
it returns the header processing by default

Comment (tonybase) [reply transport/http/codec.go:28]:
```
errors.BadRequest("CODEC", fmt.Sprintf("unregister Content-Type: %s", r.Header.Get("Content-Type")))
```
You can set it in message to provide details of the error.

Comment (workman-Lu) [reply transport/http/codec.go]: nice

File: transport/http/codec.go [hunk 3]
```diff
@@ -35,7 +35,7 @@ func DefaultRequestDecoder(r *http.Request, v interface{}) error {
 		return nil
 	}
 	if err = codec.Unmarshal(data, v); err != nil {
-		return errors.BadRequest("CODEC", err.Error())
+		return errors.BadRequest("CODEC",fmt.Sprintf("body unmarshal", err.Error()))
```
Comment (tonybase) [inline transport/http/codec.go:38]: %s

**Review by tonybase — APPROVED** [review]

**Review by tonybase — APPROVED** [review]

**Pull request merged**